### PR TITLE
Add BLAS++

### PR DESCRIPTION
**Pull request by ax3l**
Add BLAS++: modern C++ bindings for BLAS libraries.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] ~If static libraries are linked in, the license of the static library is packaged.~
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ax3l** [comment]
cc @G-Ragghianti @mgates3

I am adding this for our WarpX desktop users as mentioned per e-mail. No need to get active from your side, but if anyone of you wants to co-maintain this at any point in the future do not hesitate to reach out and we'll add you.

File: recipes/blaspp/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    - make           # [unix]
+    - ninja          # [win]
+    - cmake >=3.15.0
+    - llvm-openmp    # [osx or linux]
+  host:
+    - openblas
+    - openblas >=0.3.7 = openmp_*  # [osx or linux]
+  run:
+    - libopenblas
+    - libopenblas >=0.3.7 = openmp_*  # [osx or linux]
```
Comment (isuruf) [inline recipes/blaspp/meta.yaml:42]: Why not use the blas mechanism mentoned in our docs?

Comment (ax3l) [reply recipes/blaspp/meta.yaml:42]: Oh... https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

Comment (ax3l) [reply recipes/blaspp/meta.yaml]: Hm, that did not go as planned.

Comment (ax3l) [reply recipes/blaspp/meta.yaml]: I am really not as smart about compile matrices in CF and jinja rules as people might think :sweat:

File: recipes/blaspp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make           # [unix]
+    - ninja          # [win]
+    - cmake >=3.15.0
+    - llvm-openmp    # [osx or linux]
+  host:
+    - {{ blas_impl }}
+  run:
+    - libblas * *{{ blas_impl }}
```
Comment (ax3l) [inline recipes/blaspp/meta.yaml:40]: Copied from docs.

Warning:
```
Using .* with relational operator is superfluous and deprecated and will be removed in a future version of conda
```

File: recipes/blaspp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    - {{ blas_impl }}
```
Comment (isuruf) [inline recipes/blaspp/meta.yaml:41]:
```suggestion
  host:
    - libblas
    - libcblas
```

Comment (ax3l) [reply recipes/blaspp/meta.yaml:41]: Are those OpenMP enabled?

Comment (isuruf) [reply recipes/blaspp/meta.yaml:41]: There are openmp enabled variants and there are ones without. Using this in host enables the use of any BLAS backend at install time by the user. (This is similar to `update-alternatives` on debian)

File: recipes/blaspp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    - blaspp {{ version }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - make           # [unix]
+    - ninja          # [win]
+    - cmake >=3.15.0
+    - llvm-openmp    # [osx or linux]
```
Comment (isuruf) [inline recipes/blaspp/meta.yaml:36]:
```suggestion
    - llvm-openmp    # [osx]
    - libgomp            # [linux]
```

Comment (ax3l) [reply recipes/blaspp/meta.yaml:36]: okili dokili, using that combination already downstream.

File: recipes/blaspp/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    - llvm-openmp    # [osx]
+    - libgomp        # [linux]
+  host:
+    - libblas
+    - libcblas
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libblaspp.so                                          # [linux]
+    - otool -L ${PREFIX}/lib/libblaspp.dylib                                      # [osx]
```
Comment (isuruf) [inline recipes/blaspp/meta.yaml:45]:
```suggestion
    - test -f ${PREFIX}/lib/libblaspp${SHLIB_EXT}                                          # [unix]
```

**Review by isuruf — APPROVED** [review]

**Comment by ax3l** [comment]
Thank you for your expertise & help, @isuruf! :tada:

**Pull request merged**